Frameworks/Foundation/NSOperation.mm

```diff
@@ -77,41 +61,88 @@ + (BOOL)automaticallyNotifiesObserversForKey:(NSString*)key {
     return NO;
 }
 
-/**
- @Status Interoperable
-*/
-+ (id)allocWithZone:(NSZone*)zone {
-    NSOperation* ret = [super allocWithZone:zone];
+- (void)_checkReady {
+    bool newReady = YES;
 
-    ret->priv = new NSOperationPriv();
+    // If cancelled, skip this logic and set _ready to YES. Otherwise check dependencies.
+    if (![self isCancelled]) {
+        _dependenciesLock.lock();
+        int count = [_dependencies count];
```

Comment:
I'm upgrading this from a nit -- we should definitely use `lock_guard` and `unique_lock` everywhere. `std::condition_variable_any` actually can take a `unique_lock`, as well.

---
In reply to: [69623694](https://github.com/Microsoft/WinObjC/pull/602#discussion_r69623694) [](ancestors = 69623694)